src/Microsoft.Health.Fhir.Core/Models/BundleResourceContextComparer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Hl7.Fhir.Rest;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    /// <summary>
+    /// Compares instances of <see cref="BundleResourceContext"/> to support the correct sequence of resources.
+    /// </summary>
+    public sealed class BundleResourceContextComparer : IComparer<BundleResourceContext>
+    {
+        private static HTTPVerb[] _verbExecutionOrder = new HTTPVerb[] { HTTPVerb.DELETE, HTTPVerb.POST, HTTPVerb.PUT, HTTPVerb.PATCH, HTTPVerb.GET, HTTPVerb.HEAD };
```

Comment:
Do we plan to have this order documented anywhere? I feel it is important that customers understand the order their requests will be processed in.